backend/httpclient/config/response_limit_middleware.go

```diff
@@ -10,25 +10,21 @@ import (
 // ResponseLimitMiddlewareName is the middleware name used by ResponseLimitMiddleware.
 const ResponseLimitMiddlewareName = "response-limit"
 
-func ResponseLimitMiddleware(limit int64) httpclient.Middleware {
+func ResponseLimitMiddleware() httpclient.Middleware {
 	return httpclient.NamedMiddlewareFunc(ResponseLimitMiddlewareName, func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
 		return httpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
 			res, err := next.RoundTrip(req)
 			if err != nil {
 				return nil, err
 			}
 
-			reqLimit := limit
-			if reqLimit <= 0 {
-				reqLimit = backend.GrafanaConfigFromContext(req.Context()).ResponseLimit()
-			}
-
-			if reqLimit <= 0 {
+			limit := backend.GrafanaConfigFromContext(req.Context()).ResponseLimit()
```

Comment:
Sorry, but I find this more awkward see my latest comment. Only take the config as argument allows easier testing and cleaning separation of concerns.

Reply:
got it what you mean, updated.